src/Cli/dotnet/Commands/Test/VSTest/TestCommand.cs

```diff
@@ -298,11 +298,16 @@ internal static int RunArtifactPostProcessingIfNeeded(string testSessionCorrelat
 
     private static bool ContainsBuiltTestSources(string[] args)
     {
-        foreach (string arg in args)
+        for (int i = 0; i < args.Length; i++)
         {
-            if (!arg.StartsWith("-") &&
-                (arg.EndsWith(".dll", StringComparison.OrdinalIgnoreCase) || arg.EndsWith(".exe", StringComparison.OrdinalIgnoreCase)))
+            string arg = args[i];
+            if (arg.EndsWith(".dll", StringComparison.OrdinalIgnoreCase) || arg.EndsWith(".exe", StringComparison.OrdinalIgnoreCase))
             {
+                var previousArg = i > 0 ? args[i - 1] : null;
+                if (previousArg != null &&  CommonOptions.PropertiesOption.Aliases.Contains(previousArg))
```

Comment:
This approach will probably not work for https://github.com/dotnet/sdk/issues/51249 as the dll isn't provided in `-p`. We probably need to get a proper fix soon and get approval for getting a fix for GA?